dev/devicelab/lib/framework/runner.dart

```diff
@@ -125,3 +127,33 @@ Future<VMIsolateRef> _connectToRunnerIsolate(int vmServicePort) async {
     }
   }
 }
+
+Future<void> cleanupSystem() async {
+  print('\n\nCleaning up system after task...');
+  final String javaHome = await findJavaHome();
+  if (javaHome != null) {
+    print('\nTelling Gradle to shut down (JAVA_HOME=$javaHome)');
+    final String gradlewBinaryName = Platform.isWindows ? 'gradlew.bat' : 'gradlew';
+    final Directory tempDir = Directory.systemTemp.createTempSync('flutter_devicelab_shutdown_gradle.');
+    recursiveCopy(new Directory(path.join(flutterDirectory.path, 'bin', 'cache', 'artifacts', 'gradle_wrapper')), tempDir);
+    copy(new File(path.join(path.join(flutterDirectory.path, 'packages', 'flutter_tools'), 'templates', 'create', 'android.tmpl', 'gradle', 'wrapper', 'gradle-wrapper.properties')), new Directory(path.join(tempDir.path, 'gradle', 'wrapper')));
+    if (!Platform.isWindows) {
+      await exec(
+        'chmod',
+        <String>['a+x', path.join(tempDir.path, gradlewBinaryName)],
+        canFail: true,
+      );
+    }
+    await exec(
+      path.join(tempDir.path, gradlewBinaryName),
+      <String>['--stop'],
+      environment: <String, String>{ 'JAVA_HOME': javaHome },
+      workingDirectory: tempDir.path,
+      canFail: true,
```

Comment:
If we're allowing this to fail, do we want to log if it does (so if the issue doesn't seem to be fixed, we can check that it wasn't failing here)?

Reply:
exec() always dumps the exit code, iirc

Reply:
If I'm reading it correctly, it looks silent when `canFail = true`:

```dart
final int exitCode = await process.exitCode;

if (exitCode != 0 && !canFail)
  fail('Executable failed with exit code $exitCode.');

return exitCode;
```

Reply:
It's printed by startProcess; see line 238.

Reply:
Whoops, I thought that was a framework function. All good, thanks! :-)